File: backend/app/endpoints/rag.py
from fastapi import APIRouter, UploadFile, File, Form
from ..schema.message import ChatRequest, ChatResponse
from ..services.rag import RagService

router = APIRouter()
rag_service = RagService()

@router.post("/upload-pdf")
async def upload_pdf(file: UploadFile = File(...), session_id: str = Form(...)):
    if not file.filename.endswith('.pdf'):
        return {"error": "アップロードされたファイルはPDFではありません"}
    
    content = await file.read()
    await rag_service.process_pdf(content, file.filename, session_id)
    return {"message": "PDFが正常にアップロードされました"}

@router.post("/chat", response_model=ChatResponse)
async def rag_chat(request: ChatRequest):
    return await rag_service.get_chat_response(request)

@router.post("/clear/{session_id}")
async def clear_session_data(session_id: str):
    """特定のセッションのRAGデータをクリアする"""
    rag_service.clear_session_data(session_id)
    return {"message": "セッションのRAGデータがクリアされました"}

File: frontend/src/api/rag.ts
import { ChatRequest, ChatResponse } from '../types/chat';

export const uploadPdf = async (file: File, sessionId: string): Promise<void> => {
  const formData = new FormData();
  formData.append('file', file);
  formData.append('session_id', sessionId);

  const response = await fetch('http://localhost:8000/rag/upload-pdf', {
    method: 'POST',
    body: formData,
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(error.detail || 'PDFのアップロードに失敗しました');
  }

  return response.json();
};

export const sendRagMessage = async (request: ChatRequest): Promise<ChatResponse> => {
  const response = await fetch('http://localhost:8000/rag/chat', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(request),
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(error.detail || 'メッセージの送信に失敗しました');
  }

  return response.json();
};